### PR TITLE
docs: HOCON spec compliance checklist (S1-S26, 209 items)

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

Adds the per-implementation HOCON spec compliance tracker for ts.hocon under [`docs/spec-compliance.md`](docs/spec-compliance.md). Inherits the 209 item IDs from the canonical template in [o3co/xx.hocon](https://github.com/o3co/xx.hocon/blob/main/docs/spec-checklist.md) and adds ts.hocon-specific `tests:` and `status:` cells.

- Item count: **209**
- Status distribution: ✅ 114 / ⚠️ 1 / ❌ 7 / 🤷 67 / ➖ 20
- Compliance rate: **spec-total 54.8%** / **in-scope 60.6%**

Also updates [`.gitignore`](.gitignore) so hand-written docs at `docs/<file>.md` can be tracked without un-ignoring TypeDoc's generated output (replaces `docs/` with `docs/*` plus `!docs/spec-compliance.md`).

Notable findings surfaced by the checklist (verified by Codex round-2 review):

- ⚠️ **S8.1** (unquoted forbidden chars): cited test covers only a subset; `isUnquotedStart` / `isUnquotedContinue` in `src/internal/lexer/lexer.ts` do not exclude `` ` `` (backtick), so the impl allows it in unquoted strings contrary to spec L245.
- ❌ **S3.4** ([#55](https://github.com/o3co/ts.hocon/issues/55) still OPEN): unbraced root + stray `}` accepted. The existing test covers braced-root double-`}` (a different code path).
- ❌ **S8.6** (unquoted strings cannot begin with 0-9 or -): lexer permits digits / `-` as unquoted starts and parser turns `123abc` / `-foo` into strings rather than rejecting.
- ❌ **S13c.1–S13c.5** (`\${X[]}` env-var list): substitution lexer rejects `[` / `]` inside `\${...}` body. Feature unparseable.
- 67 🤷 items remain as test debt, concentrated in S5 comma error cases, S6 Unicode whitespace details, S10 concat error cases, S13/S13a substitution edge cases, S15 numeric-indexed object → array (entire section uncovered), S17 type-conversion error cases.

20 items inherited from the template as globally out-of-scope: classpath (JVM-only), URL include (READMEs declare unsupported), Period Format (mirrors `java.time.Period`), reference.conf / application.conf (JVM convention), MIME type, surrogate codepoint (intentional language-natural divergence), `.properties` multi-line/Unicode escape simplification.

The companion cross-implementation roll-up lives at [xx.hocon/docs/compliance-matrix.md](https://github.com/o3co/xx.hocon/blob/main/docs/compliance-matrix.md).

## Test plan

- [x] Item count: 209 (matches template)
- [x] ID order: exact match with template (`diff` empty)
- [x] All ➖ items inherit the template's `out-of-scope:` rationale verbatim
- [x] `npm test` all green (381 passed, 1 intentionally skipped)
- [x] Codex review round 2 applied (S8.1, S8.6, S13c.1-5, S20, header)
- [ ] Reviewer spot-check on a section of choice
- [ ] Decide whether to file follow-up issues for the test-debt clusters

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/spec-compliance.md [hunk 1]
```diff
@@ -0,0 +1,764 @@
+# HOCON Spec Compliance — ts.hocon
+
+This file extends the canonical item definitions in [`xx.hocon/docs/spec-checklist.md`](../../xx.hocon/docs/spec-checklist.md). It inherits all 209 items in the same order, adding `tests:` and `status:` fields for this implementation.
```
[the remark this answers is not in the document]
Comment (y1o1) [reply docs/spec-compliance.md]: Fixed in 5a557a7 — replaced the relative path with an absolute `github.com/o3co/xx.hocon` URL.

File: docs/spec-compliance.md [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+
+- **`tests:`** — path to the test or fixture exercising each item, or `—` when no test covers it (test debt).
+- **`status:`** — uses the glyphs defined in the template legend (✅ ⚠️ ❌ 🤷 ➖). Default is 🤷 (no test, unverified).
+- **Compliance rate** — computed as `(✅ + ⚠️·0.5) / total` (spec-total) and `(✅ + ⚠️·0.5) / (total − ➖)` (in-scope). See the template for the full convention. Current rollup (execution pass): ✅ 115 / ⚠️ 0 / ❌ 1 / 🤷 77 / ➖ 16 — spec-total 55.0% (115/209), in-scope 59.6% (115/193).
```
[the remark this answers is not in the document]
Comment (y1o1) [reply docs/spec-compliance.md]: Fixed in 5a557a7 — removed the inline status rollup (which would drift on every status change) and replaced it with a link to `compliance-matrix.md`, the authoritative cross-impl roll-up.
